src/consensus/malachite/host.rs

```diff
@@ -161,11 +164,31 @@ impl Host {
                 certificate,
                 consensus: consensus_ref,
             } => {
+                let proposed_value = state
+                    .shard_validator
+                    .get_proposed_value(&certificate.value_id)
+                    .unwrap();
+
+                let commits = Commits::from_commit_certificate(&certificate);
                 //commit
+                state.shard_validator.decide(commits.clone()).await;
+
+                let decided_value = if let Some(block) = proposed_value.block(commits.clone()) {
+                    Some(decided_value::Value::Block(block))
+                } else if let Some(shard_chunk) = proposed_value.shard_chunk(commits.clone()) {
+                    Some(decided_value::Value::Shard(shard_chunk))
+                } else {
+                    None
+                };
+
                 state
-                    .shard_validator
-                    .decide(Commits::from_commit_certificate(&certificate))
-                    .await;
+                    .gossip_tx
+                    .send(GossipEvent::BroadcastDecidedValue(proto::DecidedValue {
```

Comment:
Can we not re-use the shard decision tx/rx channels that the mempool uses?

Reply:
Sounds good, I'm gonna defer this because though the shard_decision_rx is available here, it requires extra work to handle it in the gossip code.